AudioKitSynthOne/DSP/Kernel/S1DSPKernel.hpp

```diff
@@ -49,9 +47,12 @@ class S1DSPKernel : public AKSoundpipeKernel, public AKOutputBuffered {
     ~S1DSPKernel();
 
     // public accessor for protected sp
-    sp_data *spp() {
+    inline sp_data *spp() {
         return sp;
     }
+    inline int sampleRate() {
```

Comment:
I suggest instead of using the Soundpipe Kernel, introduce a member variable in `S1DSPKernel.hpp` called `mSampleRate` and set it to the SampleRate when the Kernel's `init` function is called (to avoid co-dependence on the AKSoundpipeKernel).

Reply:
👍

Reply:
Hi @sieren, i assume you would get rid of the getter to speed up the sample rate lookup? 

@all, my objection to @sieren 's solution would be that you'd be introducing a second source of truth for the sample rate and therefore the possibility for diverging sample rates and adding to the confusion of future contributors ("hm, I think we need to change mSampleRate. Oh well, now everything sounds out of tune...").
In my opinion, this drawback should be justified with a measurable speedup.

Reply:
Hey, yeah, youre totally right. Ignore my comment for now :) Even though I think there won't be changes to the sampleRate without the DSP Kernel being completely reset (and reinitialized) it's probably best to have it in one place. Might be worth refactoring this a little bit in a future PR.
@aure what do you think?